### ISSUE TITLE
Problem with preventing click on anchor ("a"), bug?

### ISSUE THREAD

**Issue by saturday1** [mattbryson/TouchSwipe-Jquery-Plugin#272]
Hello!

I have a jQuery script that does this (I have slimmed it down a bit to make it more understandable):

```javascript
swipeLeft:function(event, direction, distance, duration, fingerCount, fingerData) {
	$(this).find('a').click(function(e) {
		e.preventDefault();
	});
	thumbs.goToNextSlide();
},
swipeRight:function(event, direction, distance, duration, fingerCount, fingerData) {
	$(this).find('a').click(function(e) {
		e.preventDefault();
	});
	thumbs.goToPrevSlide();
},

```

It is running on a slideshow, and I have removed "a" from "excludedElements". 

So what do I want the script to do? Since each slide is an anchor with image I want to prevent click when swiping. This works this far: After reload, I can click the slide to go to the link. I can swipe images without firing the link. But after 2 swipes, I can not click the slide any longer. It is prevented, even if no swipe.

Test:
1. Click on a slide (Link should work)
2. Swipe one time, click the slide (link should work)
3. Swipe more then two times (Link does not fire anymore)

In other words, I want the link to fire on click/tap. But if you swipe I ofc do not want the link/anchor to fire.

Is there a way to get rid of the "bug"? To test this, look at: http://cdn.axiell.com/dev/slippry/index.html

**Comment by saturday1** [comment]
Tried this, still same result though:

```javascript
if(direction == 'right' || direction == 'left') {
	jQuery(this).find('a').click(function(e) {
		e.preventDefault();
	});
} else {
	jQuery(location).href(jQuery(this).find('a'));
};
```

**Comment by saturday1** [comment]
Also tried this:

```javascript
var thisLink = jQuery(this).find('a');
var thisLinkHref = jQuery(this).find('a').attr('href');

if(direction == 'right' || direction == 'left') {
	console.log(direction);
	thisLink.click(function(e) {
		e.preventDefault();
	});
} else {
	thisLink.unbind('click');
	jQuery(location).attr('href',thisLinkHref);
};
```

**Comment by mattbryson** [comment]
Sorry for late reply on this one.   The plugin supports binding its own tap / click event, so you can differentiate between swiping and taping.

http://labs.rampinteractive.co.uk/touchSwipe/demos/Tap_vs_swipe.html
